Add connection between bot and converter

marl0vv opened the pull request:
Сделал пока одну функцию, чтобы глянуть, как оно будет работать.
Добавил контроллер, который делает запрос к конвертеру и получает статус конвертации
Добавил боту команду для использования функции из контроллера.
Перенёс из конвертера dto и enums.
Добавил exception

marl0vv committed:
Add BotController

Create controller that connects bot-service and converter-service.
Add function getTaskStatus as the easiest in realization to test how it works.

marl0vv committed:
Add Dto and enums

In order for controller to work we need to add dto and enums. A duplication from WebpConverter service but it's okay for now I guess.

marl0vv committed:
Add conversionStatus ability to a bot

Add an ability to a bot that allows to ask for an image conversion status through telegram bot.

marl0vv committed:
Add ConversionTaskNotFoundException

Catch exception when conversion task is not found in webpconverter and then throw new exception that is caught in a bot and in a handler for requests

RavenVSS commented on src/main/java/com/glamik/converter_bot/ConverterBot.java:
```diff
@@ -25,11 +32,18 @@ public class ConverterBot extends AbilityBot implements SpringLongPollingBot {
     private final String botToken;
 
+    @Value("${bot.creatorId}")
+    private long creatorId;
```
А этот конфиг нормально задаётся?

RavenVSS commented on src/main/java/com/glamik/converter_bot/controller/ConverterBotController.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @GetMapping("/bot/{taskId}/status")
+    public ConversionTaskStatusDto getTaskStatus(@PathVariable UUID taskId) {
```
В боте не должно быть контроллера, тут должен быть FeignClient

https://www.baeldung.com/spring-cloud-openfeign

Т.е. создаёшь интерфейс сервиса с методами куда хочешь кидать запросы и оно автоматом создаётся. Далее этот интерфейс внедряешь куда надо и вызываешь методы

RavenVSS commented on src/main/java/com/glamik/converter_bot/enums/ErrorMessage.java:
```diff
@@ -0,0 +1,6 @@
+package com.glamik.converter_bot.enums;
```
На сколько я помню, snake-case для пакетов не рекомендуется  

Можешь почитать соглашение по наименованиям  
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

RavenVSS commented on src/main/java/com/glamik/converter_bot/ConverterBot.java:
```diff
@@ -53,11 +67,60 @@ public LongPollingUpdateConsumer getUpdatesConsumer() {
+                .build();
+    }
+
+    public Ability conversionStatus() {
```
Получается команда должна быть в таком формате? 

`/status 00000000-0000-0000-0000-00000000`

marl0vv committed:
Change package name from snake_case to lowercase

So Oracle documentation doesn't really prohibit underscores but Google conventions does. There is no unanimous opinion  on this problem really but I think that its okay to go with lowercase

marl0vv committed:
Change controller to feign

RavenVSS commented on src/main/java/com/glamik/converterbot/exception/ExceptionApiHandler.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice(annotations = RestController.class)
+public class ExceptionApiHandler {
```
Уже не нужен, т.к. обработка ошибок в feign по другому происходит

marl0vv committed:
Delete user exceptions

Delete exception handler and user exceptions because of change from controller to feign

marl0vv committed:
Refactor getStatus

Didn't like how it took null so switched to Optional. Looks better.
